### PR TITLE
Fix wb_robot_set_mode argument type

### PR DESCRIPTION
**Pull request by stefaniapedrazzi**
Address #844: change argument type of `wb_robot_set_mode` from `void *` to `const char *` so that it can be handled correctly also in the Java and Python APIs.

 I also tried to make the argument name consistent between the code and the documentation.
And added `wb_robot_set_mode` function to the MATLAB API.

### CODE REVIEW

**Comment by stefaniapedrazzi** [comment]
:+1: :heavy_check_mark: (travis_unit)

**Comment by DavidMansolino** [comment]
:eyes:

File: docs/reference/changelog-r2020.md [hunk 1]
```diff
@@ -29,6 +29,7 @@ Released on December YYth, 2019.
     - Added a [script to cleanup the Webots preferences](https://github.com/cyberbotics/webots/blob/master/scripts/preferences_cleaner/README.md).
     - Linux: Added support for Python 3.8.
   - Enhancements
+    - **Updated argument type in [`wb_robot_set_mode()`](robot.md#wb_robot_set_mode) function from `void *` to `const char *`.**
```
Comment (DavidMansolino) [inline docs/reference/changelog-r2020.md]:
```suggestion
    - **Updated argument type in the [`wb_robot_set_mode()`](robot.md#wb_robot_set_mode) function from `void *` to `const char *`.**
```

That sound better  no?

**Comment by stefaniapedrazzi** [comment]
:+1:

**Pull request merged**